azurerm/internal/services/databricks/databricks_workspace_resource_test.go

```diff
@@ -279,31 +349,558 @@ resource "azurerm_databricks_workspace" "test" {
+//   resource_group_name = azurerm_resource_group.test.name
+//   location            = azurerm_resource_group.test.location
+//   sku                 = "%[3]s"
+
+//   custom_parameters {
+//     enable_cmk_encryption = true
+//   }
+// }
+// `, data.RandomInteger, data.Locations.Primary, sku, data.RandomString)
+// }
+
+// func (DatabricksWorkspaceResource) customerManagedKeysStepTwo(data acceptance.TestData, sku string) string {
+// 	return fmt.Sprintf(`
+// provider "azurerm" {
+//   features {}
+// }
+
+// data "azurerm_client_config" "current" {}
+
+// resource "azurerm_resource_group" "test" {
+//   name     = "acctestRG-db-%[1]d"
+//   location = "%[2]s"
+// }
+
+// resource "azurerm_key_vault" "test" {
+//   name                = "acctest-kv-%[4]s"
+//   location            = azurerm_resource_group.test.location
+//   resource_group_name = azurerm_resource_group.test.name
+//   tenant_id           = data.azurerm_client_config.current.tenant_id
+//   sku_name            = "premium"
+
+//   purge_protection_enabled = true
+// }
+
+// resource "azurerm_key_vault_key" "test" {
+//   depends_on = [azurerm_key_vault_access_policy.test]
+
+//   name         = "acctest-key-%[1]d"
+//   key_vault_id = azurerm_key_vault.test.id
+//   key_type     = "RSA"
+//   key_size     = 2048
+
+//   key_opts = [
+//     "decrypt",
+//     "encrypt",
+//     "sign",
+//     "unwrapKey",
+//     "verify",
+//     "wrapKey",
+//   ]
+// }
+
+// resource "azurerm_key_vault_access_policy" "test" {
+//   key_vault_id = azurerm_key_vault.test.id
+//   tenant_id    = azurerm_key_vault.test.tenant_id
+//   object_id    = data.azurerm_client_config.current.object_id
+
+//   key_permissions = [
+//     "get",
+//     "list",
+//     "create",
+//     "decrypt",
+//     "encrypt",
+//     "sign",
+//     "unwrapKey",
+//     "verify",
+//     "wrapKey",
+//     "delete",
+//     "restore",
+//     "recover",
+//     "update",
+//     "purge",
+//   ]
+// }
+
+// resource "azurerm_key_vault_access_policy" "databricks" {
+//   key_vault_id = azurerm_key_vault.test.id
+//   tenant_id    = azurerm_databricks_workspace.test.storage_account_identity.0.tenant_id
+//   object_id    = azurerm_databricks_workspace.test.storage_account_identity.0.principal_id
+
+//   key_permissions = [
+//     "get",
+//     "unwrapKey",
+//     "wrapKey",
+//     "delete",
+//     "purge",
+//   ]
+// }
+
+// resource "azurerm_databricks_workspace" "test" {
+//   name                = "acctestDBW-%[1]d"
+//   resource_group_name = azurerm_resource_group.test.name
+//   location            = azurerm_resource_group.test.location
+//   sku                 = "%[3]s"
+
+//   custom_parameters {
+//     enable_cmk_encryption = true
+
+//     customer_managed_key {
+//       key_source    = "Microsoft.KeyVault"
+//       key_name      = azurerm_key_vault_key.test.name
+//       key_version   = azurerm_key_vault_key.test.version
+//       key_vault_uri = azurerm_key_vault.test.vault_uri
```

Comment:
could we not all get these from the key_vault_key id?

Reply:
Not sure, this might be one of those chicken or the egg scenarios... I will investigate if this is possible.